### ISSUE TITLE
Optimize baseline experience (before progressive enhancement)

### ISSUE THREAD

**Issue by tylersticka** [cloudfour/cloudfour.com-patterns#1096]
As of this writing, IE usage has dipped to 1.13% globally ([source](https://gs.statcounter.com/)). It no longer makes sense to use IE as the "lowest common denominator" of our site's experience, preventing us from embracing what features like CSS Grid Layout, Custom Properties, Service Workers, etc. can offer us.

That said, we also believe fundamentally in accessibility and progressive enhancement. There's no reason for our content to be inaccessible to older browsers.

So our goal should _not_ be to closely approximate the appearance or behavior of our experiences in <= IE11, but to make sure we're fencing the right CSS in `@supports` blocks and the right JavaScript behind feature checks. The experience for IE11 and earlier users should ideally be that of a modestly styled but well-formed HTML document.

(More detail and to-dos will be added once we're in a good place to start this effort and can appraise what's already acceptable versus areas of improvement.)

**Comment by tylersticka** [comment]
Based on our conversation today, we're going to address issues that directly impact the most common unsupported browsers. Our goal is to make information accessible, not to "fix" the design or bring it in line with that of modern browsers.

Based on some research, I think the browsers that would yield the best results to focus on would be IE11 and Opera Mini.

**Comment by tylersticka** [comment]
## Critical IE11 Issues

Buttons and other themed elements do not appear to have defaults/fallbacks set, resulting in white on gray or white on white text:

<img width="167" alt="Screen Shot 2022-05-24 at 3 26 51 PM" src="https://user-images.githubusercontent.com/69633/170142564-8b240248-d05c-4b68-80ef-764296970057.png">

Our heading component is not sized correctly, possibly due to there being no fallback for `clamp` being unsupported. When the permalink function is available, it overlaps the text.

<img width="319" alt="Screen Shot 2022-05-24 at 3 28 04 PM" src="https://user-images.githubusercontent.com/69633/170142642-35cd92a2-45b3-4f81-9c8d-290cfae221f0.png">

The notifications button appears even though IE11 does not support them:

<img width="232" alt="Screen Shot 2022-05-24 at 3 27 32 PM" src="https://user-images.githubusercontent.com/69633/170142675-f47ef4fa-f710-4d41-a13c-d0938438f19f.png">

On the home page, the book image overlays adjacent content:

<img width="301" alt="Screen Shot 2022-05-24 at 3 26 44 PM" src="https://user-images.githubusercontent.com/69633/170142740-a9cdb12c-4d0d-47f0-a140-31e2bf39aceb.png">

**Comment by tylersticka** [comment]
## Opera Mini Issues

These are mostly similar to IE11 in that they mostly seem related to a lack of support for custom properties or `clamp`. But in addition to the IE issues, there are issues with "Menu" text partially covering nav items and the footer illustration covering footer content:

![IMG_4836](https://user-images.githubusercontent.com/69633/170143379-1e1b903d-8461-4bd8-b5b2-6d6e04cf7011.PNG)

![IMG_4835](https://user-images.githubusercontent.com/69633/170143392-ef72ae84-0568-475d-870e-06c590c63c04.PNG)

Also the "clients we've worked with" logos are invisible:

![IMG_4837](https://user-images.githubusercontent.com/69633/170143471-fb1ccec4-2e22-4d92-8d6e-6efdb9ac4684.PNG)

**Comment by tylersticka** [comment]
## TL;DR

- When we use custom properties, we should probably include a fallback value.
- When we use `clamp`, we should probably include a fallback value.
- We should adjust a couple of image sizes and/or layouts, either fencing them more strictly behind `@supports` or making sure a maximum dimension is set.

**Comment by calebeby** [comment]
@tylersticka
> The notifications button appears even though IE11 does not support them

The "Get weekly digests" button functionality currently requires JS. Our current JS situation doesn't work in IE11. We _could_ add a `nomodule` fallback but that would be quite a bit of work.

Do we want a fallback experience (e.g. just display the email signup form directly) for when JS doesn't work?

**Comment by tylersticka** [comment]
> Do we want a fallback experience (e.g. just display the email signup form directly) for when JS doesn't work?

That sounds like a good idea to me.

**Comment by calebeby** [comment]
I tried opening up https://c4dev.wpengine.com/ on my phone in opera mini. It seems to be working fine? It does not at all look like the screenshots you shared. I must be doing something wrong? It looks the same as in Chrome on my phone 😕

**Comment by calebeby** [comment]
Oh I see there is a setting in the app to change the data savings amount. The default works like a normal browser, but if I set it to "extreme" then it says "some sites may not work as expected" and then it looks like your screenshot.

It feels to me like this might not be a use-case worth spending time fixing, unless there is an easy way for me to test out local changes without pushing to wp engine

**Comment by tylersticka** [comment]
@calebeby Opera Mini has 1.08% global usage compared to IE11's 0.57% and is still actively developed for Android.

I know it isn't very convenient to test in, but since IE11 is desktop-only, I think it's a good example of making sure our work is resilient enough to still be accessible in Opera Mini.

**Comment by calebeby** [comment]
> @calebeby Opera Mini has 1.08% global usage compared to IE11's 0.57% and is still actively developed for Android.
> I know it isn't very convenient to test in, but since IE11 is desktop-only, I think it's a good example of making sure our work is resilient enough to still be accessible in Opera Mini.

Opera mini does not by default come with the setting turned on that breaks our site. By default, our site works fine. If you turn on the "extreme savings" mode _then_ it breaks the site. Do you know if the percentages you are using are based on the number of Opera Mini users with the "extreme savings" setting turned on, or whether they are based on the total number of Opera mini users?

**Comment by calebeby** [comment]
I am fine with testing in Opera mini, but I do not have a process for doing so. Should I set up a new WP engine environment so I can make changes and test them? Or push directly to the c4dev environment?

**Comment by spaceninja** [comment]
@calebeby you should be able to access your local machine from your phone as long as they're on the same network by using your IP: https://stackoverflow.com/questions/3132105/how-do-you-access-a-website-running-on-localhost-from-iphone-browser

For example, my local machine has the IP 192.168.1.4, and when I go to 192.168.1.4:3000 on my iPhone, it loads my local dev site.

**Comment by Paul-Hebert** [comment]
In addition to Scott's suggestion, if you wanted to test a local copy of the WP site, you can use the Local Live Links feature (though it requires setting up a free account)

https://localwp.com/live-links/

FWIW I've seen some weird PHP errors when using this feature but it may still be helpful

**Comment by calebeby** [comment]
Thanks, live links works great!

**Comment by tylersticka** [comment]
> By default, our site works fine. If you turn on the "extreme savings" mode _then_ it breaks the site. Do you know if the percentages you are using are based on the number of Opera Mini users with the "extreme savings" setting turned on, or whether they are based on the total number of Opera mini users?

The "extreme savings" mode is Opera Mini's big differentiator in regions without fast, reliable and/or affordable internet options. [caniuse](https://caniuse.com/usage-table) separates "Opera Mobile" (the Opera browser running on mobile) from "Opera Mini" (you can tell because browser versions aren't really a factor when the page is rendered on a server, hence Mini's all-encompassing "all" version).

The ask is less about "I'm worried Opera Mini users won't be able to access the site," and more about finding reasonable test cases for making our content resilient enough to work in less capable browsers. If you can find me a mobile browser that fits the bill better than Opera Mini with a higher usage, I can definitely consider it.

And obviously, all we can do is our best. If there are insurmountable hurdles, those won't be launch blockers. But I was able to get our homepage animation to fallback to a static image on Opera Mini, so I'm confident we can find solutions for the most egregious problems if we put our minds to it!

**Comment by calebeby** [comment]
Thanks Tyler, that makes a lot of sense.

The IE11 fixes seemed to fix all the issues you pointed out in Opera Mini as well 🎉
